hazelcast/src/main/java/com/hazelcast/query/SqlPredicate.java

```diff
@@ -163,63 +164,104 @@ private Predicate createPredicate(String sql) {
                     if ("=".equals(token) || "==".equals(token)) {
                         int position = (i - 2);
                         validateOperandPosition(position);
-                        Object first = toValue(tokens.remove(position), mapPhrases);
-                        Object second = toValue(tokens.remove(position), mapPhrases);
-                        setOrAdd(tokens, position, equal((String) first, (Comparable) second));
+                        String first = (String) toValue(tokens.remove(position), mapPhrases);
```

Comment:
This repeating logic should be simplified.

Reply:
Can you work on removing the duplication? At least for the newly added parts.